world/console.cpp

```diff
@@ -962,6 +962,268 @@ void ConsoleQuit(
+		return;
+	}
+
+	const auto& type = Strings::ToLower(args[0]);
+
+	const auto is_character  = type == "character";
+	const auto is_expedition = type == "expedition";
+	const auto is_group      = type == "group";
+	const auto is_guild      = type == "guild";
+	const auto is_name       = type == "name";
+	const auto is_raid       = type == "raid";
+
+	if (
+		!is_character &&
+		!is_expedition &&
+		!is_group &&
+		!is_guild &&
+		!is_name &
+		!is_raid
+	) {
+		connection->SendLine("czmove character [character_id] [instance_id]");
+		connection->SendLine("czmove character [character_id] [zone_short_name]");
+		connection->SendLine("czmove expedition [expedition_id] [instance_id]");
+		connection->SendLine("czmove expedition [expedition_id] [zone_short_name]");
+		connection->SendLine("czmove group [group_id] [instance_id]");
+		connection->SendLine("czmove group [group_id] [zone_short_name]");
+		connection->SendLine("czmove guild [guild_id] [instance_id]");
+		connection->SendLine("czmove guild [guild_id] [zone_short_name]");
+		connection->SendLine("czmove name [character_name] [instance_id]");
+		connection->SendLine("czmove name [character_name] [zone_short_name]");
+		connection->SendLine("czmove raid [raid_id] [instance_id]");
+		connection->SendLine("czmove raid [raid_id] [zone_short_name]");
+		return;
+	}
+
+	const auto    update_identifier = !is_name ? Strings::ToInt(args[1]) : 0;
+	const auto&   name              = is_name ? Strings::UcFirst(Strings::ToLower(args[1])) : std::string();
+	const auto&   zone_short_name   = !Strings::IsNumber(args[2]) ? args[2] : "";
+	const uint16  instance_id       = Strings::IsNumber(args[2]) ? static_cast<uint16>(Strings::ToUnsignedInt(args[2])) : 0;
+
+	uint8 update_type;
+
+	if (is_character) {
+		update_type = CZUpdateType_Character;
+	} else if (is_expedition) {
+		update_type = CZUpdateType_Expedition;
+	} else if (is_group) {
+		update_type = CZUpdateType_Group;
+	} else if (is_guild) {
+		update_type = CZUpdateType_Guild;
+	} else if (is_name) {
+		update_type = CZUpdateType_ClientName;
+	} else if (is_raid) {
+		update_type = CZUpdateType_Raid;
+	}
+
+	auto pack = new ServerPacket(ServerOP_CZMove, sizeof(CZMove_Struct));
+	auto* CZM = (CZMove_Struct*) pack->pBuffer;
+
+	CZM->update_type       = update_type;
+	CZM->update_subtype    = !instance_id ? CZMoveUpdateSubtype_MoveZone : CZMoveUpdateSubtype_MoveZoneInstance;
+	CZM->update_identifier = update_identifier;
+	CZM->instance_id       = instance_id;
+
+	strn0cpy(CZM->zone_short_name, zone_short_name.c_str(), sizeof(CZM->zone_short_name));
+	strn0cpy(CZM->client_name, name.c_str(), sizeof(CZM->client_name));
+
+	zoneserver_list.SendPacket(pack);
+	safe_delete(pack);
+
+	connection->SendLine(
+		fmt::format(
+			"Moving player(s) to {} by {} with an identifier of {}.",
+			!instance_id ? zone_store.GetZoneLongName(zone_store.GetZoneID(zone_short_name), true) : fmt::format("Instance ID {}", instance_id),
```

Comment:
This works, but fyi you can use the zone store to fetch the entire zone object by the short name

```cpp
ZoneRepository::Zone *GetZone(const std::string& in_zone_name);
```

```cpp
ZoneRepository::Zone z = zone_store.GetZone(zone_short_name);
// access via z->member
```

Reply:
Also makes it easier to validate when the zone doesn't exist

```cpp
if (z->id == 0) {
    // return validation error
}
```